cefdatabrowse/cefdatabrowse_support.py

```diff
@@ -0,0 +1,521 @@
+    <xsl:template match="/">
+        <xsl:apply-templates mode="%s"/>
+    </xsl:template>
+    %s
+</xsl:stylesheet>'''
+
+
+class FileResolver(etree.Resolver):
+    _path = None
+
+    def __init__(self, path):
+        self._path = path
+        pass
+
+    def resolve(self, url, pubid, context):
+        if url.startswith('http://'):
+            return self.resolve_filename(url, context)
+        else:
+            return self.resolve_filename(os.path.abspath(self._path + '/' + url), context)
+
+    pass
+
+
+def application(filename, params):
+    """ Entry Point for CEFDatabrowse Application """
+    try:
+        starttime = time()
+        import databrowse.support.cef_web_support as db_web_support_module
+
+        # Set up web_support class with environment information
+        db_web_support = db_web_support_module.web_support(filename, params)
+
+        # Determine Requested File/Folder Absolute Path and Path Relative to Dataroot
+        if "path" not in db_web_support.req.form:
+            fullpath = db_web_support.dataroot
+            relpath = '/'
+            pass
+        else:
+            fullpath = '/'.join(os.path.splitdrive(os.path.abspath(db_web_support.req.form["path"].value))[1].split('\\'))
+            if not fullpath.startswith(db_web_support.dataroot):
+                return db_web_support.req.return_error(403)
+            if os.access(fullpath, os.R_OK) and os.path.exists(fullpath):
+                if fullpath == db_web_support.dataroot:
+                    relpath = '/'
+                    pass
+                else:
+                    if db_web_support.dataroot != "/":
+                        relpath = fullpath.replace(db_web_support.dataroot, '')
+                    else:
+                        relpath = fullpath
+                    pass
+                pass
+            elif not os.path.exists(fullpath):
+                return db_web_support.req.return_error(404)
+            else:
+                return db_web_support.req.return_error(401)
+            pass
+
+        # Convert urls from windows style to unix style
+        relpath = '/'.join(relpath.split('\\'))
+        fullpath = '/'.join(fullpath.split('\\'))
+
+        # Determine handler for requested path
+        import databrowse.support.handler_support as handler_support_module
+        handler_support = handler_support_module.handler_support(db_web_support.icondbpath, db_web_support.hiddenfiledbpath, db_web_support.directorypluginpath)
+        handlers = handler_support.GetHandler(fullpath)
+        handler = handlers[-1]
+
+        # Let's see if we want to override the default handler
+        if "handler" in db_web_support.req.form:
+            handler = db_web_support.req.form['handler'].value
+            pass
+
+        # Get A Handle to The Rendering Plugin
+        caller = "cefdatabrowse"
```

Comment:
Is it necessary to call the caller here "cefdatabrowse" instead of Databrowse.  This is simply intended as a way of identifying whether a plugin is being called by Databrowse directly (at the top level) or if it is being called by another plugin (or itself) recursively at a lower level.  Changing this to "databrowse" would remove the edits from most of the other plugins.  If there is a required corner case where we need to be able to tell whether we are being called by databrowse or cefdatabrowse, perhaps we set some parameter elsewhere and check that.

Reply:
I have reverted that change. The only problem that will crop up is when trying to upload files and exception will be thrown in CEFDatabrowse... at least until I find a solution to the POST/GET problem.